ctk/common/src/main/kotlin/org/codice/compliance/verification/binding/RedirectBindingVerifier.kt

```diff
@@ -260,16 +279,33 @@ class RedirectBindingVerifier(private val response: IdpRedirectResponseDecorator
     @Suppress("ComplexMethod" /* Complexity due to nested `when` is acceptable */)
     private fun decodeAndVerifyErrorResponse() {
         val samlResponse = response.samlResponse
-        val samlEncoding = response.samlEncoding
-        val decodedMessage: String
+
+        // Need to url decode SAMLEncoding first to check the encoding method uri
+        val samlEncoding = response.samlEncoding?.let {
+            try {
+                URLDecoder.decode(it, StandardCharsets.UTF_8.name())
+            } catch (e: UnsupportedEncodingException) {
+                throw SAMLComplianceException.create(SAMLBindings_3_4_4_1_c,
+                        message = "Could not url decode the SAMLEncoding parameter.",
+                        cause = e)
+            }
+        }
+
+        samlEncoding?.let {
+            if (StringUtils.isBlank(it) || !URI.create(it).isAbsolute)
```

Comment:
❓ Don't we need to catch the IllegalArgumentException if the string is not a URI?